workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,233 @@
+    environment: tech-approve
+
+  qa-approval:
+    if: ${{ github.event.inputs.environment }} != "DEV" && ${{ github.event.inputs.environment }} != "QA"
+    needs: tech-approval
+    runs-on: [ubuntu-20.04]
+    environment: qa-approve
+
+  stakeholder-approval:
+    if: ${{ github.event.inputs.environment }} == "PROD"
+    needs: [tech-approval, qa-approval]
+    runs-on: [ubuntu-20.04]
+    environment: stakeholder-approve
+
+  build-deploy:
+    needs: [tech-approval, qa-approval, stakeholder-approval]
+    runs-on: [self-hosted, windows-2019]
+    environment: ${{ github.event.inputs.environment }}
+    steps:
+      - name: Open a PagerDuty Maintenance Window
+        id: open-window
+        uses: im-open/open-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{secrets.PAGERDUTY_API_KEY}}
+          description: '${{ env.PAGERDUTY_WINDOW_DESC }}'
+          minutes: ${{ env.PAGERDUTY_WINDOW_IN_MIN }}
+          service-ids: ${{ env.PAGERDUTY_SERVICE_IDS }}
+
+      - run: |
+          echo "The maintenance window ID is: ${{ steps.open-window.outputs.maintenance-window-id }}"
+          deploy-the-code.sh
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      # Uncomment this if the site hasn't been created before
+      # - name: Create Web Site
+      #   uses: im-open/iis-site-create@v1.0.0
+      #   with:
+      #     server: '${{ secrets.iis_server }}'
+      #     website-name: '${{env.WEBSITE_NAME}}'
+      #     app-pool-name: '${{ secrets.pool-name }}'
+      #     website-host-header: '${{env.WEBSITE_HOST_HEADER}}'
+      #     website-path: '${{env.DEPLOYMENT_FOLDER}}'
+      #     website-cert-path: '${{ env.WEBSITE_CERT_PATH}}'
+      #     website-cert-password: '${{ secrets.site_cert_password }}'
+      #     website-cert-friendly-name: '${{env.WEBSITE_CERT_FRIENDLY_NAME}}'
+      #     service-account-id: '${{secrets.iis_admin_user}}'
+      #     service-account-password: '${{secrets.iis_admin_password}}'
+      #     server-public-key: ${{ env.IIS_SERVER_CERT_PATH}}
+
+      # Do NOT upload the DEPLOY_ZIP as an artifact after you've done this!
+      # DEPLOY_ZIP will contain all of your sensitive values!
+      # If an artifact is uploaded, only upload ARTIFACT_ZIP
+      - name: Update appsettings.json with Environment Specific Values
+        uses: im-open/add-or-update-json-properties@v1.0.1
+        with:
+          path-to-json-file: '${{ env.PROJECT_ROOT }}/appsettings.json'
+          # TODO:  add any appsettings.json items that need to be updated for the environment.  Format:
+          # {"SecretName": "SecretValues"},
+          # {"Nested.SecretName": "${{secrets.NESTED_SECRET}}"},
+          # {"Auth0.ClientId": "${{ secrets.AUTH0_CLIENT_ID }}"}
+          properties-to-update-or-add: |
+            [
+            ]
```

Comment:
It seems like they might need multiple of these steps, one for each environment. Am I right in that thinking? If so, should we make note of that in a comment?

Reply:
It might not just be secrets though, this probably needs a note to indicate they should only use env variables that have been set according to the environment or secrets inside of this thing or to repeat this block for each env if it is easier.  In one of account management's workflows they repeated this block for each env because it was easier than all of the env variables they had to set.